WooCommerce/WordPressAuthenticator/Unified Auth/View Related/Login/MagicLinkRequestViewController.swift

```diff
@@ -0,0 +1,230 @@
+
+        headerStackView.layer.cornerRadius = 8
+        headerStackView.layer.borderWidth = 1
+        headerStackView.layer.borderColor = UIColor.systemGray3.cgColor
+
+        return headerStackView
+    }
+
+    private func createFallbackButton() -> UIButton {
+        let fallbackButton = NUXButton()
+        fallbackButton.contentInsets = .zero
+        fallbackButton.buttonStyle = .linkButtonStyle
+        fallbackButton.customizeFont(.preferredFont(forTextStyle: .body))
+
+        fallbackButton.setTitle(fallbackButtonTitle(), for: .normal)
+        fallbackButton.on(.touchUpInside) { [weak self] _ in
+            self?.openFallbackScreen()
+        }
+        return fallbackButton
+    }
+
+    private func createPrimaryButton() -> UIButton {
+        let primaryButton = NUXButton()
+        primaryButton.isPrimary = true
+        submitButton = primaryButton
+        primaryButton.setTitle(Localization.sendMagicLink, for: .normal)
+        primaryButton.on(.touchUpInside) { [weak self] _ in
+            self?.sendMagicLink()
+        }
+
+        return primaryButton
+    }
+
+    private func pinSubviewToHorizontalEdges(_ subView: UIView) {
+        subView.translatesAutoresizingMaskIntoConstraints = false
+        subView.leadingAnchor.constraint(equalTo: stackView.leadingAnchor, constant: stackView.layoutMargins.left).isActive = true
+        subView.trailingAnchor.constraint(equalTo: stackView.trailingAnchor, constant: -stackView.layoutMargins.right).isActive = true
+    }
+}
+
+private extension MagicLinkRequestViewController {
+    struct Localization {
+        static let description = NSLocalizedString(
+            "login.magicLinkRequest.description",
+            value: "We'll email you a link that'll log you in instantly, no password needed.",
+            comment: "Description text for the magic link request form."
+        )
+        static let sendMagicLink = NSLocalizedString(
+            "login.magicLinkRequest.sendMagicLink",
+            value: "Send link by email",
+            comment: "Button title to send the magic link."
+        )
+        static let passwordFallback = NSLocalizedString(
+            "login.magicLinkRequest.passwordFallback",
+            value: "Use your password instead",
+            comment: "Button title to fallback to password login."
+        )
+        static let wpcomUsernamePasswordFallback = NSLocalizedString(
+            "login.magicLinkRequest.wpcomUsernamePasswordFallback",
+            value: "Use username and password instead",
+            comment: "Button title to fallback to WordPress.com username and password login."
+        )
+        static let magicLinkError = NSLocalizedString(
+            "login.magicLinkRequest.error",
+            value: "Something went wrong. Please try again.",
+            comment: "Error message when magic link request fails."
+        )
+    }
+
+    private func fallbackButtonTitle() -> String {
```

Comment:
```suggestion
    func fallbackButtonTitle() -> String {
```

nit: This method is already inside a private extension.